effect_player.cpp

```diff
@@ -39,7 +62,7 @@ namespace bnb::oep
 
     /* effect_player::effect_player CONSTRUCTOR */
     effect_player::effect_player(const std::vector<std::string>& path_to_resources, const std::string& client_token)
-        : m_utility(path_to_resources, client_token)
```

Comment:
it might be better to do it like this:
effect_player::effect_player(const std::vector<std::string>& path_to_resources, const std::string& client_token)
{
    // create the utility_manager
    static std::vector<std::string> s_path_to_resources;
    if (!s_utility) {
        s_path_to_resources = path_to_resources;
        s_utility = std::make_unique<bnb::utility>(path_to_resources, client_token);
    } else {
        assert(s_path_to_resources == path_to_resources);
    }
    // create the effect_player
    m_ep = bnb::interfaces::effect_player::create({
            1, // fx_width - the effect's framebuffer width
            1, // fx_height - the effect's framebuffer height
            bnb::interfaces::nn_mode::automatically,
            bnb::interfaces::face_search_mode::good,
            false,
            false});
}